doc/src/manual/arrays.md

```diff
@@ -1028,33 +1028,57 @@ It is sometimes useful to perform element-by-element binary operations on arrays
 sizes, such as adding a vector to each column of a matrix. An inefficient way to do this would
 be to replicate the vector to the size of the matrix:
 
-```julia-repl
-julia> a = rand(2,1); A = rand(2,3);
+```jldoctest broadcasting
+julia> a = [1, 2]; A = [10 20 30; 40 50 60]
 
-julia> repeat(a,1,3)+A
-2×3 Array{Float64,2}:
- 1.20813  1.82068  1.25387
- 1.56851  1.86401  1.67846
+julia> repeat(a, 1, 3) + A
+2×3 Matrix{Int64}:
+ 11  21  31
+ 42  52  62
 ```
 
-This is wasteful when dimensions get large, so Julia provides [`broadcast`](@ref), which expands
-singleton dimensions in array arguments to match the corresponding dimension in the other array
+This is wasteful when dimensions get large, so Julia provides
+[`broadcast`](@ref), which aligns dimensions in array arguments from left to
+right[^2], fills lacking dimensions with a singleton dimension, expands
+singleton dimensions to match the corresponding dimension in the other array
 without using extra memory, and applies the given function elementwise:
 
-```julia-repl
+[^2]: This is the opposite direction of that of NumPy, which aligns dimensions
+from right to left.
+
+```jldoctest broadcasting
 julia> broadcast(+, a, A)
-2×3 Array{Float64,2}:
- 1.20813  1.82068  1.25387
- 1.56851  1.86401  1.67846
+2×3 Matrix{Int64}:
+ 11  21  31
+ 42  52  62
 
-julia> b = rand(1,2)
-1×2 Array{Float64,2}:
- 0.867535  0.00457906
+julia> b = [10 20]
+1×2 Matrix{Int64}:
+ 10  20
 
 julia> broadcast(+, a, b)
-2×2 Array{Float64,2}:
- 1.71056  0.847604
- 1.73659  0.873631
+2×2 Matrix{Int64}:
+ 11  21
+ 12  22
+```
+
+In the example above, the sizes of array `a` and `A` are `(2,)` and `(2, 3)`,
```

Comment:
Would prefer `size(a) == (2,)` to this "respectively" story, to emphasise that this is a function's name